src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -58,11 +58,23 @@ def create_ramp_fit_class(model, dqflags=None, suppress_one_group=False):
     else:
         dark_current_array = model.average_dark_current
 
+    orig_gdq = None
+    wh_chargeloss = np.where(np.bitwise_and(model.groupdq.astype(np.uint32), dqflags['CHARGELOSS']))
```

Comment:
I have noticed in testing the previous version of ramp fit that this `where` call takes a ton of memory to complete, maybe because of the forced casting to np.uint32.  Since this is only used to decide whether to copy the groupdq here, can we find a less expensive substitute?  And maybe only call it if algorithm is OLS_C?

Reply:
I can research this.  Maybe `np.any` can be used in conjunction with `np.bitwise_and` to be faster than `np.where`.  Also, I will see if this is necessary.  The copy is done because the arrays get modified in during ramp fitting.  However, the way the C-extension is written, the code operates on local copies, possibly making this copy unnecessary.  In the mean time I conditioned it on running the `OLS_C` algorithm.